content/essentials/understanding-jsx.mdx

```diff
@@ -0,0 +1,105 @@
+import { Aside } from "~/components/configurable/Aside";
+
+# Understanding JSX
+
+JSX is a syntax extension for JavaScript. 
+It allows you to write HTML-like code inside your JavaScript file which keeps your rendering logic and content in the same place. 
+This provides a concise and readable way to create and represent components.
+
+## How Solid uses JSX
+
+Solid was designed to align closely with HTML standards. 
+It offers a distinct advantage: copy/paste solutions from resources like Stack Overflow, and direct usage of templates from design tools.
+
+What sets Solid's apart is the ability to use JSX immediately as it returns DOM elements. 
+This adds dynamic expressions that allow you to reference variables and functions within your HTML by using the `{ }` syntax:
+
+```jsx
+const animal = { breed: "cat", name: "Midnight" };
+const element = <p>I have a {animal.breed} named {animal.name}!</p>;
+```
+
+This allows for the use of JavaScript content in the rendered output based on the application's state or logic.
+
+Additionally, Solid's [reactive](/essentials/intro-to-reactivity) system introduces [fine-grained reactivity](/advanced/understanding-fine-grained-reactivity) with JSX.
+This ensures that only the necessary portions of the [DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction) are updated when changes occur in the underlying state, optimizing efficiency and overall performance.
+
+## Using JSX in Solid
+
+### 1. Return a single root element
+
+Unlike HTML, where you can have disconnected tags at the top level, JSX requires that a component's render method returns a single root element.
+
+<Aside type="advanced">
+**Note**: When working with JSX, the system translates parts of your code into structured HTML that's placed at the start of the file.
+Static elements, which don't change, are processed differently from dynamic ones, which might change based on data or user actions. 
+For dynamic elements, special markers are added for better handling during rendering.
+<br/>
+Having a single root ensures a consistent and manageable hierarchy, optimizing rendering and updates.
+</Aside>
+
+With its structured approach, JSX maintains the familiar nested, tree-like structure found in HTML. 
+Consequently, parent-child relationships between elements are intuitive and easy to understand.
+
+### 2. Close all tags
+
+Self-closing tags are a must in JSX. 
+Unlike in HTML, where elements like `<input>`, `<img>`, or `<br>` don't require explicit closure, JSX requires consistent self-closing tags. 
+This ensures uniformity and helps avoid potential rendering issues.
+
+### 3.  Properties vs. attributes
+
+At first glance, HTML attributes and JSX props may seem simlar, but they serve different purposes and behave differently.
```

Comment:
```suggestion
At first glance, HTML attributes and JSX properties may seem similar, but they serve different purposes and behave differently.
```

Reply:
> JSX props

Remember what I said about jargon/shorthand. Explain it the first time you mention it. You do so later in the document, so write it out fully here.